### PR TITLE
cmd: tikv-ctl upgrade to futures 0.3

### PR DESCRIPTION
**Pull request by ekexium**
### What problem does this PR solve?

Problem Summary:

### What is changed and how it works?

Change tikv-ctl.rs to use futures 0.3

### Check List 

Tests 

- No code

### CODE REVIEW

**Comment by hunterlxt** [comment]
LGTM

File: cmd/src/bin/tikv-ctl.rs [hunk 1]
```diff
@@ -643,22 +642,18 @@ impl DebugExecutor for DebugClient {
...
-        Box::new(
+        Box::pin(
             self.scan_mvcc(&req)
                 .unwrap()
+                .compat()
                 .map_err(|e| e.to_string())
-                .map(|mut resp| (resp.take_key(), resp.take_info())),
-        ) as Box<dyn Stream<Item = (Vec<u8>, MvccInfo), Error = String>>
+                .map_ok(|mut resp| (resp.take_key(), resp.take_info())),
+        ) as MvccInfoStream
```
Comment (sticnarf) [inline cmd/src/bin/tikv-ctl.rs:656]: It seems that this `as` is unnecessary?

File: cmd/src/bin/tikv-ctl.rs [hunk 2]
```diff
@@ -817,17 +812,12 @@ impl DebugExecutor for Debugger {
...
-        limit: u64,
-    ) -> Box<dyn Stream<Item = (Vec<u8>, MvccInfo), Error = String>> {
+    fn get_mvcc_infos(&self, from: Vec<u8>, to: Vec<u8>, limit: u64) -> MvccInfoStream {
         let iter = self
             .scan_mvcc(&from, &to, limit)
             .unwrap_or_else(|e| perror_and_exit("Debugger::scan_mvcc", e));
-        let stream = stream::iter_result(iter).map_err(|e| e.to_string());
-        Box::new(stream) as Box<dyn Stream<Item = (Vec<u8>, MvccInfo), Error = String>>
+        let stream = stream::iter(iter).map_err(|e| e.to_string());
+        Box::pin(stream) as MvccInfoStream
```
Comment (sticnarf) [inline cmd/src/bin/tikv-ctl.rs:820]: ditto

File: cmd/src/bin/tikv-ctl.rs [hunk 3]
```diff
@@ -341,15 +348,16 @@ trait DebugExecutor {
 
                 let mut take_item = |i: usize| -> Option<(Vec<u8>, MvccInfo)> {
                     let wait = match i {
-                        1 => future::poll_fn(|| mvcc_infos_1.poll()).wait(),
-                        _ => future::poll_fn(|| mvcc_infos_2.poll()).wait(),
+                        1 => block_on(future::poll_fn(|cx| mvcc_infos_1.poll_next_unpin(cx))),
+                        _ => block_on(future::poll_fn(|cx| mvcc_infos_2.poll_next_unpin(cx))),
                     };
                     match wait {
```
Comment (sticnarf) [inline cmd/src/bin/tikv-ctl.rs:354]: I think you can `match wait?` to avoid `Some`/`None` matches.

File: cmd/Cargo.toml [hunk 4]
```diff
@@ -58,6 +58,7 @@ engine_rocks = { path = "../components/engine_rocks" }
 engine_traits = { path = "../components/engine_traits" }
 fs2 = "0.4"
 futures = "0.1"
```
Comment (nrc) [inline cmd/Cargo.toml:60]: do we still need the futures 0.1 dep?

Comment (sticnarf) [reply cmd/Cargo.toml:60]: the tikv-server bin still needs that

**Pull request merged**